plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java

```diff
@@ -187,7 +188,15 @@ public void setParameters(List<IParameter> parameters) {
 		  parameters.forEach(p -> paramsMap.put(p.getName(), p.getValue()));
 		}
 		originalValueMap = paramsMap;
-		firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = injectProjectParameters(this.eclipseProject, parameters));
+		/*
+		 * firePropertyChange will send an event if the array are differents so we need to check
+		 * otherwise we will risk the ObservableList to be based on the wrong list as it is assigned
+		 * in the call.
+		 */
+		List<IParameter> newParameters = injectProjectParameters(this.eclipseProject, parameters);
+		if (!Objects.equals(this.parameters, newParameters)) {
+	        firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = newParameters);
```

Comment:
If we want to be strict and not call the fire, then in each similar case we should wrap the code with 

if (!Objects.equals(this.list, newList)) {
   //
}

We will not gain much, because when the list is changed, the comparison of lists will be performed twice - here and in pcs. My point is that we should treat firing lists consistently in cases that have the same pattern. If calling equals() seems good, it should be inserted everywhere we fire lists. Also, since it is good to ensure the same list instance in the model and clients, it may be done in this case too, no matter if equality check is done or not. Otherwise if we decide that ensuring the same list is not needed, then it should be removed from all other cases, for the sake of consistency.